src/integration/billing/BillingIntegration.ts

```diff
@@ -364,6 +366,28 @@ export default abstract class BillingIntegration<T extends BillingSetting> {
         detailedMessages: { error: error.message, stack: error.stack }
       });
     }
+    const tenant = await TenantStorage.getTenant(this.tenantID);
```

Comment:
I would expect the code above (line 351 -> 368 to not be in that method but in the one which create the invoice.
Here I want only to see the code to send the invoice that's it.

Reply:
Moved it into the concrete billing implementation, as the invoice finalization depends on how the billing provider works.
In this case with Stripe, not at the invoice creation but at the state switch DRAFT -> OPEN